editor/modes/visual-editor/block.js

```diff
@@ -129,14 +131,26 @@ class VisualEditorBlock extends wp.element.Component {
 		);
 	}
 
-	componentDidUpdate() {
+	componentDidUpdate( prevProps ) {
 		if ( this.previousOffset ) {
 			window.scrollTo(
 				window.scrollX,
 				window.scrollY + this.node.getBoundingClientRect().top - this.previousOffset
 			);
 			this.previousOffset = null;
 		}
+
+		// Focusing the block node if no inner element is focused
+		if (
+			!! this.props.focus &&
+			! prevProps.focus &&
+			(
+				! document.activeElement ||
+				document.activeElement.closest( '.editor-visual-editor__block' ) !== this.node
```

Comment:
The logic no longer exists, but this could have been expressed more simply as:

```
! this.node.contains( document.activeElement )
```

https://developer.mozilla.org/en-US/docs/Web/API/Node/contains